indra_gpt/run_statement_json_extraction.py

```diff
@@ -148,89 +203,111 @@ def main(json_file, model: str, n_iter: int, output_file: Path, verbose: bool =
-        json.loads and stmt_from_json functions to return a statement
-        json. Some are not. Use the try/except method to extract statement
-        jsons from the generated json objects for the ones that work.
-        Append them to the list of all statements extracted. If it doesn't
-        work on a generated json object, just append that there was an
-        error loading the statement.
-        """
-        try:
-            # Here run json.loads on the response separately from stmt_from_json
-            # to clearly see if the response is a valid json object
-
-            # JSON loads the response
-            stmt_json = json.loads(response)
-
-            # Run post-processing on the extracted json
-            stmt_json = post_process_extracted_json(stmt_json)
-
-            # Extract the INDRA statement object from the json
-            stmt_n = stmt_from_json(stmt_json)
-
-            # Append the str extracted statement to statements list (while not saved
-            # as a valid statement in the output TSV file, this gives a quick
-            # view of the statement extracted when loading the file later)
-            statements.append(str(stmt_n))
-        except (JSONDecodeError, IndexError, TypeError) as e:
-            # TypeError can happen when the response contains more than one
-            # statement json object and the post-processing function tries to
-            # access the "type" key in the response from the list of statement
-            # json objects.
-            # Todo: handle multiple statement json objects in a response
-            statements.append(f"Error: {e}")
-
-    # Save sentences, json_object_list, outputs, and statements as a pandas dataframe
-    df = pd.DataFrame(
-        {
-            "sentence": sentences,
-            "input": [json.dumps(js) for js in json_object_list],
-            "generated_json_object": outputs,
-            "extracted_statement": statements,
-        }
-    )
-    # Save dataframe as tsv file
-    df.to_csv(output_file, sep="\t", index=False)
+    json_object_map = create_json_object_map(json_object_list)
+
+    if batch_jobs:
+        sentences = []
+        sample_lists = []
+        json_objects = []
+        statements = []
+
+        for matches_hash in json_object_map:
+            json_object = json_object_map[matches_hash]
+            json_objects.append(json_object)
+
+            sentence = json_object["evidence"][0]["text"]
+            sentences.append(sentence)
+
+            sequence = list(set(json_object_map.keys()) - {matches_hash})
+            sample_hashes = random.sample(sequence, 2)
+            sample_list = [json_object_map[h] for h in sample_hashes]
+            sample_lists.append(sample_list)
+
+        batch_id = gpt_stmt_json_batch(
+            sample_lists, json_objects, model=model, debug=verbose
+        )
+        logger.info(f"Batch job submitted with ID: {batch_id}")
```

Comment:
There should be a way to save the batch job id somewhere so that a later script can read that and check that ID.